### DISCUSSION TITLE
CCefClientDelegate::GetScreenInfo 崩溃

### DISCUSSION THREAD

**Discussion by sig999** [CefView/QCefView#115] [Q&A] [answered]
CCefClientDelegate::GetScreenInfo
 QWindow* pWindow = dynamic_cast<QWindow*>(pCefViewPrivate_->q_ptr->window()); 
Window时碰到了指针，不知道是为何。
修改如下之后可以正常运行
  if (!pWindow) {
    // the view is not visible so we retrieve the main screen info
    currentScreen = QApplication::screens().at(0);

  } else {
        currentScreen = pWindow->screen();
    if (!currentScreen) {
      // the view is not visible so we retrieve the main screen info
      currentScreen = QApplication::screens().at(0);
    }
  }

**Comment by tishion** [comment, marked as answer]
多谢你发现的问题，不过你的修复方法有问题。
```cpp
QWindow* pWindow = dynamic_cast<QWindow*>(pCefViewPrivate_->q_ptr->window());
```
我不知道这是从哪里来的代码，项目中是没有这么写的。
这里有两个错误：
1.冗余的使用dynamic_cast，因为整个项目都没有开启C++的RTTI特性，所以dynamic cast在这里毫无作用，跟static cast是一样的。
2. QWidget::window()方法返回的是***QWidget****类型，无论是static cast 还是dynamic cast都无法转换成功，因为QWindow和QWidget没有继承关系。


为了避免极端情况下的空指针访问，我增加了一层检查，修复：#116
